xla/stream_executor/cuda/cuda_blas_lt.cc

```diff
@@ -131,33 +138,31 @@ tsl::Status BlasLt::Init() {
 }
 
 /*static*/ tsl::StatusOr<BlasLt::MatrixLayout> BlasLt::MatrixLayout::Create(
-    blas::DataType type, size_t num_rows, size_t num_cols,
-    BlasLt::MatrixLayout::Order order, size_t batch_size,
-    std::optional<int64_t> leading_dim_stride,
-    std::optional<int64_t> batch_stride) {
-  if (!leading_dim_stride) {
-    leading_dim_stride = (order == Order::kRowMajor) ? num_cols : num_rows;
-  }
+      const gpu::MatrixLayout& m) {
 
+  TF_ASSIGN_OR_RETURN(auto type, gpu::AsBlasDataType(m.dtype));      
+
+  // if (!leading_dim_stride) {
+  //   leading_dim_stride = (order == Order::kRowMajor) ? num_cols : num_rows;
+  // }
   cublasLtMatrixLayout_t cu_layout;
   SE_CUBLAS_RETURN_IF_ERROR(
-      cublasLtMatrixLayoutCreate(&cu_layout, AsCudaDataType(type), num_rows,
-                                 num_cols, *leading_dim_stride));
+      cublasLtMatrixLayoutCreate(&cu_layout, AsCudaDataType(type), m.num_rows,
+                                 m.num_cols, m.leading_dim_stride));
   // Wrap cublas handle immediately, so it is cleaned up if an error occurs.
   BlasLt::MatrixLayout layout(cu_layout);
   TF_RETURN_IF_ERROR(
       SetAttr(cu_layout, CUBLASLT_MATRIX_LAYOUT_ORDER,
-              int32_t{(order == Order::kRowMajor) ? CUBLASLT_ORDER_ROW
-                                                  : CUBLASLT_ORDER_COL}));
+          int32_t{(m.order == gpu::MatrixLayout::Order::kRowMajor) ? 
+                CUBLASLT_ORDER_ROW : CUBLASLT_ORDER_COL}));
   TF_RETURN_IF_ERROR(SetAttr(cu_layout, CUBLASLT_MATRIX_LAYOUT_BATCH_COUNT,
-                             static_cast<int32_t>(batch_size)));
-
-  if (!batch_stride) {
-    batch_stride = (batch_size > 1) ? num_rows * num_cols : 0;
-  }
+                             static_cast<int32_t>(m.batch_size)));
 
+  // if (!batch_stride) {
```

Comment:
Same here, please either delete this code (plus possibly add a TODO), or uncomment